cdap-api/src/main/java/co/cask/cdap/api/dataset/lib/PartitionedFileSetArguments.java

```diff
@@ -0,0 +1,153 @@
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.dataset.lib;
+
+import co.cask.cdap.api.annotation.Beta;
+import co.cask.cdap.api.dataset.lib.Partitioning.FieldType;
+
+import java.util.Map;
+import javax.annotation.Nullable;
+
+/**
+ * Helpers for manipulating runtime arguments of time-partitioned file sets.
+ */
+@Beta
+public class PartitionedFileSetArguments {
+
+  public static final String OUTPUT_PARTITION_KEY_PREFIX = "output.partition.";
+  public static final String INPUT_PARTITION_LOWER_PREFIX = "input.filter.lower.";
+  public static final String INPUT_PARTITION_UPPER_PREFIX = "input.filter.upper.";
+  public static final String INPUT_PARTITION_VALUE_PREFIX = "input.filter.value.";
+
+  /**
+   * Set the partition key  of the output partition when using PartitionedFileSet as an OutputFormatProvider.
+   * This key is used as the partition key for the new file, and also to generate an output file path - if that path
+   * is not explicitly given as an argument itself.
+   *
+   * @param key the partition key
+   * @param arguments the runtime arguments for a partitioned dataset
+   */
+  public static void setOutputPartitionKey(Map<String, String> arguments, PartitionKey key) {
+    for (Map.Entry<String, ? extends Comparable> entry : key.getFields().entrySet()) {
+      arguments.put(OUTPUT_PARTITION_KEY_PREFIX + entry.getKey(), entry.getValue().toString());
+    }
+  }
+
+  /**
+   * @return the partition key of the output partition to be written; or null if no partition key was found
+   *
+   * @param arguments the runtime arguments for a partitioned dataset
+   * @param partitioning the declared partitioning for the dataset, needed for proper interpretation of values
+   */
+  @Nullable
+  public static PartitionKey getOutputPartitionKey(Map<String, String> arguments, Partitioning partitioning) {
+    // extract the arguments that describe the output partition key
+    Map<String, String> keyArguments = FileSetProperties.propertiesWithPrefix(arguments, OUTPUT_PARTITION_KEY_PREFIX);
+    if (keyArguments.isEmpty()) {
+      return null; // there is no output partition key
+    }
+    // there is a partition key; now it is required to match the partitioning
+    PartitionKey.Builder builder = PartitionKey.builder();
+    for (Map.Entry<String, FieldType> entry : partitioning.getFields().entrySet()) {
+      String fieldName = entry.getKey();
+      FieldType fieldType = entry.getValue();
+      String stringValue = keyArguments.get(fieldName);
+      Comparable fieldValue = convertFieldValue("key", "value", fieldName, fieldType, stringValue, false);
+      @SuppressWarnings({ "unchecked", "unused" }) // we know it's type safe, but Java does not
+      PartitionKey.Builder unused = builder.addField(fieldName, fieldValue);
```

Comment:
I can only suppress warnings on a declaration, hence using a temp variable. Then I have to add "unused" to the suppressed warnings, otherwise I get a warning about an unused. Java sometimes makes it hard to make your code free of warnings :)